archive_updater.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#still working through a couple of things and haven't tested everything, but this is a rough draft of the changes to come
+
+import subprocess
+import os
+
+def update_archive(archive_path:str):
+    subprocess.check_call(['git', '-C', archive_path,'fetch'])
+    subprocess.check_call(['git', '-C', archive_path,'pull'])
+    
+def fill_message_directory(archive_path:str, directory_path:str, last_used_commit_hash:str) -> str:
+    process = subprocess.Popen(['git', '-C', archive_path, 'log', '--format=format:%H'], stdout=subprocess.PIPE)
+    output = process.communicate()[0]
+    message_hashes = output.decode('utf-8').split()
+    
+    file_count = 0
+    while(message_hashes[file_count] != last_used_commit_hash):
+        file = os.path.join(directory_path, 'email' + str(file_count) + '.txt')
+        with open(file, 'w+') as output:
```

Comment:
hmm, do we need `w+`, aka read+write on this file?
I would have thought just `'w'` would have been enough.